public/app/features/explore/Logs/Logs.tsx

```diff
@@ -669,31 +683,32 @@ const UnthemedLogs: React.FunctionComponent<Props> = (props: Props) => {
       updateItem(logsParent.id, { expanded: true });
     }
 
-    register?.({
-      icon: 'gf-logs',
-      title: 'Pinned log',
-      panelId: 'Logs',
-      level: 'child',
-      ref: null,
-      color: LogLevelColor[row.logLevel],
-      childOnTop: true,
-      onClick: () => onOpenContext(row, () => {}),
-      onRemove: (id: string) => {
-        unregister?.(id);
-        if (getPinnedLogsCount() < PINNED_LOGS_LIMIT) {
-          setPinLineButtonTooltipTitle('Pin to content outline');
-        }
-      },
-    });
+    const alreadyPinned = pinnedLogs?.find((pin) => pin.id === row.rowId);
+    if (alreadyPinned && row.rowId && allowUnPin) {
+      unregister?.(row.rowId);
+    } else if (getPinnedLogsCount() !== PINNED_LOGS_LIMIT) {
+      register?.({
+        id: row.rowId,
+        icon: 'gf-logs',
+        title: PINNED_LOGS_TITLE,
+        panelId: 'Logs',
```

Comment:
Can we introduce a const for the value of `Logs` and use that also where we find the relevant items in the `outlineItems`?

Reply:
Done